app/controllers/api/v1/meetings_controller.rb

```diff
@@ -81,25 +83,32 @@ def find_room
         @room = Room.find_by!(friendly_id: params[:friendly_id])
       end
 
-      def authorized_as_viewer?(access_code:)
-        return true if access_code.blank?
+      def authorized_as_viewer?(viewer_code:)
+        return true if viewer_code.blank? && params[:access_code].blank?
 
-        access_code == params[:access_code]
+        viewer_code == params[:access_code]
       end
 
-      def authorized_as_moderator?(access_code:)
-        (params[:access_code].present? && access_code == params[:access_code]) ||
-          @room.anyone_joins_as_moderator? ||
+      # Five scenarios where a user is authorized to join a BBB meeting as a moderator
+      # The user joins
+      # 1. its own room
+      # 2. a shared room
+      # 3. a room that requires a moderator access code and the access code input is correct
+      # 4. a room that has the AnyoneJoinAsModerator setting toggled on and does not require an access code
+      # 5. a room that has the AnyoneJoinAsModerator setting toggled on and requires a moderator or a viewer access code
+      #    and the access code input correspond to either the moderator or the viewer access code
```

Comment:
Umm, I thought about the scenarios from a different perspective:
      # The user joins
      # 1. its own room
      # 2. a shared room
      # 3. a room with AnyoneJoinAsModerator setting off but has a moderator access code that the user input matched. 
      # 4. a room that has the AnyoneJoinAsModerator setting toggled on and does not require an access code (Here we don't care about the moderator access code).
      # 5. a room that has the AnyoneJoinAsModerator setting toggled on and requires an access code
      #    which the user input correspond to.
      The moderator access code is optional, it has a meaning only if the AnyoneJoinAsModerator setting if off it enables users to join in as moderators.
      When the option is toggled on the optional code has no effects.
      The access code (viewer access code) is required and its the true access code that if enabled will define the eligibility of joining or not a meeting.
      We may update the UI so that the moderator access code will not be displayed when the setting is toggled on.
      We have some other mutual exclusive settings that we haven't touched yet.

Reply:
We can discuss different persepctives and see the pros and cons of each from the users expectations,